lib/ansible/executor/task_executor.py

```diff
@@ -682,10 +688,13 @@ def _evaluate_failed_when_result(result):
                 result['attempts'] = retries - 1
                 result['failed'] = True
 
+        if self._task.action not in C._ACTION_WITH_CLEAN_FACTS:
```

Comment:
instead of these 2 changes just move the `variables[self._task.register] = result = wrap_var(result)` to line 700